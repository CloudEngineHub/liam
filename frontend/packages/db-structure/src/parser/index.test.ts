import fs from 'node:fs'
import path from 'node:path'
import { describe, expect, it } from 'vitest'
import { parse } from '.'

describe(parse, () => {
  it('should parse schema.rb to JSON correctly', () => {
    const schemaText = fs.readFileSync(
      path.resolve(__dirname, './schemarb/input/schema1.in.rb'),
      'utf-8',
    )

    const result = parse(schemaText, 'schemarb')
    expect(result).toMatchSnapshot()
  })
})
