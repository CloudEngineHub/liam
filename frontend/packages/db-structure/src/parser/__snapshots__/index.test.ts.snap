// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parse > should parse schema.rb to JSON correctly 1`] = `
{
  "relationships": {},
  "tables": {
    "users": {
      "color": null,
      "columns": [
        {
          "check": null,
          "comment": null,
          "default": null,
          "increment": false,
          "name": "id",
          "notNull": false,
          "primary": false,
          "type": "varchar",
          "unique": false,
        },
        {
          "check": null,
          "comment": null,
          "default": null,
          "increment": false,
          "name": "company_id",
          "notNull": true,
          "primary": false,
          "type": "uuid",
          "unique": false,
        },
        {
          "check": null,
          "comment": null,
          "default": null,
          "increment": false,
          "name": "created_at",
          "notNull": true,
          "primary": false,
          "type": "datetime",
          "unique": false,
        },
        {
          "check": null,
          "comment": null,
          "default": "new user",
          "increment": false,
          "name": "name",
          "notNull": false,
          "primary": false,
          "type": "string",
          "unique": false,
        },
        {
          "check": null,
          "comment": null,
          "default": 30,
          "increment": false,
          "name": "age",
          "notNull": false,
          "primary": false,
          "type": "string",
          "unique": false,
        },
        {
          "check": null,
          "comment": null,
          "default": false,
          "increment": false,
          "name": "is_deleted",
          "notNull": false,
          "primary": false,
          "type": "string",
          "unique": false,
        },
        {
          "check": null,
          "comment": null,
          "default": null,
          "increment": false,
          "name": "user_name",
          "notNull": true,
          "primary": false,
          "type": "string",
          "unique": false,
        },
      ],
      "comment": null,
      "indices": [],
      "name": "users",
      "x": 0,
      "y": 0,
    },
  },
}
`;
